### PR TITLE
Homework1

### PR DESCRIPTION
**Pull request by Taras-Lavriv**

### CODE REVIEW

File: Homework_01/Task_5/Test/Test_maximal_subarray_sum.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from Homework_01.Task_5.find_maximal_subarray_sum import find_maximal_subarray_sum
+
+
+def test_max_sum():
+    assert find_maximal_subarray_sum([1, 3, -1, -3, 5, 3, 6, 7], 3) == 16
```
Comment (fake-commit) [inline Homework_01/Task_5/Test/Test_maximal_subarray_sum.py:5]: Please add more tests with edge cases: e.g. what if the input list is empty or k = 0.

File: Homework_01/Task_4/Check_sum_of_four.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def check_sum_of_four(a, b, c, d):
+    quantity = 0
+    for i in a:
+        for j in b:
+            for k in c:
+                for n in d:
+                    if i + j + k + n == 0:
+                        quantity += 1
```
Comment (fake-commit) [inline Homework_01/Task_4/Check_sum_of_four.py:8]: Try to reduce algorithm complexity. You have 4 nested loops, which means that the complexity of the algorithm ~ O (n * 4).

File: Homework_01/Task_4/Test/Test_Sum_of _four.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from Homework_01.Task_4.Check_sum_of_four import check_sum_of_four
+
+
```
Comment (fake-commit) [inline Homework_01/Task_4/Test/Test_Sum_of _four.py:3]: Provide more positive and negative tests.

File: Homework_01/Task_3/Tests/data1.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+1, 2, 3, 4, 5
```
Comment (fake-commit) [inline Homework_01/Task_3/Tests/data1.txt:1]: The file should contain line-delimited integers, not comma-delimited.

File: Homework_01/Task_3/Min_Max_tuple.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+def find_maximum_and_minimum(file_name: str):
+    with open(file_name) as fi:
+        for lines in fi:
+            line = lines.strip()
+            list_of_data = line.split(', ')
+            return int(min(list_of_data)), int(max(list_of_data))
```
Comment (fake-commit) [inline Homework_01/Task_3/Min_Max_tuple.py:6]: Please see the comment I left on the input file, it should contain line-delimited integers, not comma-delimited. So, you have to read the file line by line and return the minimum and maximum values after processing the entire file.

File: Homework_01/Task_2/tests/test_fib_checker.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from Homework_01.Task_2.fib_checker import check_fibonacci
+
+
+def test_positive_case():
+    """Testing that actual part of fibonacci give True"""
+    assert check_fibonacci([8, 13, 21, 34, 55, 89, 144, 233])
+
```
Comment (fake-commit) [inline Homework_01/Task_2/tests/test_fib_checker.py:7]: Please provide more tests. e.g, the input sequence could be 0 length or could contain less than 3 elements, then your function will break.

File: Homework_01/Task_2/fib_checker.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
+from typing import Sequence
+
+
+def check_fibonacci(data: Sequence[int]) -> bool:
+    flag = True
+    for i in range(len(data) - 2):
+        if data[i] + data[i + 1] == data[i + 2]:
```
Comment (fake-commit) [inline Homework_01/Task_2/fib_checker.py:7]: What if the data length is less than 3 elements?

**Review by fake-commit — CHANGES_REQUESTED** [review]

**Review by fake-commit — CHANGES_REQUESTED** [review]
Please see my comments. Also, please remove files that are not homework 1 related (.idea, homework_2, homework_3 directories).

**Review by fake-commit — APPROVED** [review]
